syntax = "proto3";

package eu.sotec;

message empty {}

message mqtt_service_config {
  string hostname = 1;
  uint32 port = 2;
  uint32 qos = 3;
  optional string username = 4;
  optional string password = 5;
}

message mqtt_message {
  string topic = 1;
  string message = 2;
  bool retained = 3;
}

service MqttService {
  rpc PublishMessage(mqtt_message) returns (empty);
}
